packages/nextjs/src/server/clerkMiddleware.ts

```diff
@@ -184,7 +197,7 @@ const parseHandlerAndOptions = (args: unknown[]) => {
   return [
     typeof args[0] === 'function' ? args[0] : undefined,
     (args.length === 2 ? args[1] : typeof args[0] === 'function' ? {} : args[0]) || {},
-  ] as [ClerkMiddlewareHandler | undefined, ClerkMiddlewareOptions];
+  ] as [ClerkMiddlewareHandler | undefined, ClerkMiddlewareOptions | ((req?: NextRequest) => ClerkMiddlewareOptions)];
```

Comment:
I'll extract `(req?: NextRequest) => ClerkMiddlewareOptions)` to a separate type to reuse on the function overload definition as well.